Tiltfile

```diff
@@ -388,6 +401,107 @@ def prepare_all():
+        fail(filename + " not found")
+
+    os.environ["NAMESPACE"] = substitutions.get("NAMESPACE", "default")
+    os.environ["KUBERNETES_VERSION"] = substitutions.get("KUBERNETES_VERSION", "v1.24.0")
+    os.environ["CONTROL_PLANE_MACHINE_COUNT"] = substitutions.get("CONTROL_PLANE_MACHINE_COUNT", "1")
+    os.environ["WORKER_MACHINE_COUNT"] = substitutions.get("WORKER_MACHINE_COUNT", "3")
+
+    basename = os.path.basename(filename)
+    if basename.endswith(".yaml"):
+        if basename.startswith("clusterclass-"):
+            template_name = basename.replace("clusterclass-", "").replace(".yaml", "")
+            deploy_clusterclass(template_name, provider, filename)
+        elif basename.startswith("cluster-template-"):
+            clusterclass_name = basename.replace("cluster-template-", "").replace(".yaml", "")
+            deploy_cluster_template(clusterclass_name, provider, filename)
+
+def deploy_clusterclass(clusterclass_name, provider, filename):
+    apply_clusterclass_cmd = "cat " + filename + " | " + envsubst_cmd + " | " + kubectl_cmd + " apply -f - && echo \"ClusterClass created from\'" + filename + "\', don't forget to delete\n\""
+    delete_clusterclass_cmd = kubectl_cmd + " delete clusterclass " + clusterclass_name + ' --ignore-not-found=true; echo "\n"'
+
+    local_resource(
+        name = clusterclass_name,
+        cmd = apply_clusterclass_cmd,
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = [provider + "-clusterclasses"],
+    )
+
+    cmd_button(
+        clusterclass_name + ":apply",
+        argv = ["sh", "-c", apply_clusterclass_cmd],
+        resource = clusterclass_name,
+        icon_name = "note_add",
+        text = "Apply ClusterClass",
+    )
+
+    cmd_button(
+        clusterclass_name + ":delete",
+        argv = ["sh", "-c", delete_clusterclass_cmd],
+        resource = clusterclass_name,
+        icon_name = "delete_forever",
+        text = "Delete ClusterClass",
+    )
+
+def deploy_cluster_template(template_name, provider, filename):
+    apply_cluster_template_cmd = "CLUSTER_NAME=" + template_name + "-$RANDOM; " + clusterctl_cmd + " generate cluster $CLUSTER_NAME --from " + filename + " | " + kubectl_cmd + " apply -f - && echo \"Cluster '$CLUSTER_NAME' created, don't forget to delete\n\""
+
+    delete_clusters_cmd = 'DELETED=$(echo "$(bash -c "' + kubectl_cmd + ' get clusters --no-headers -o custom-columns=":metadata.name"")" | grep -E "^' + template_name + '-[[:digit:]]{1,5}$"); if [ -z "$DELETED" ]; then echo "Nothing to delete for cluster template ' + template_name + '"; else echo "Deleting clusters:\n$DELETED\n"; echo $DELETED | xargs -L1 ' + kubectl_cmd + ' delete cluster; fi; echo "\n"'
+
+    local_resource(
+        name = template_name,
+        cmd = apply_cluster_template_cmd,
+        auto_init = False,
+        trigger_mode = TRIGGER_MODE_MANUAL,
+        labels = [provider + "-cluster-templates"],
+    )
```

Comment:
I guess this is one of those times where a giant command is a bit awkward to debug and I don't have an Ubuntu box I can test Tilt on. Do you know if `CLUSTER_NAME` is an empty string here? Another possibility is `$RANDOM` isn't available on every shell and `CLUSTER_NAME` ended up becoming `development-` which fails the regex, or maybe `CLUSTER_NAME` was set as an env variable somewhere else.

Reply:
I'll test it out. I think you might be right with $RANDOM. $RANDOM works on my zsh, but on sh it's just empty

Reply:
Okay works if using bash instead of sh
```
    local_resource(
        name = template_name,
        cmd = ["bash", "-c", apply_cluster_template_cmd],
        auto_init = False,
        trigger_mode = TRIGGER_MODE_MANUAL,
        labels = [provider + "-cluster-templates"],
    )
```

Given that we already have a dependency on bash in `delete_clusters_cmd` WDYT about using bash consistently instead of running some commands with sh and some others with bash? (I would also explicitly use bash for `local_resources`)

Reply:
Sure that makes sense to me!

Reply:
Thx!